specs/Freeze.md

```diff
@@ -0,0 +1,155 @@
+# Background
+Edge browser has a sleeping tab feature which reduces resource usage when the tab is in the background. We are introducing WebView2 APIs
+to access this feature so that invisible WebView can use less resources. We'd appreciate your feedback.
+
+
+# Description
+You may call the `TryFeezeAsync` API to have the WebView2 consume less memory. This is useful when your Win32 app becomes invisible, or when your Universal Windows Platform app is suspended, during the suspended event handler before completing the suspended event.
```

Comment:
To me this looks as a violation of the establish .NET naming conventions, and with this it breaks the principle of the "least surprise", as the behaviour of the proposed API would be unexpected to .NET developers.

"Try" methods are generally companion methods for non-"try" methods, e.g. `int.Parse(...)` and `int.TryParse(...)`, with the former throwing exceptions and the latter not throwing exceptions in event of a failure.
The "try" methods have the following signature `bool Try<Operation>(T input, out V value)`, with the return value indicating whether the "operation" has succeeded or not. 
Related docs: https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/exceptions-and-performance#try-parse-pattern


With this one would expect:
- `Task FreezeAsync()`, and
- `Task<bool> TryFreezeAsync()`.

But then it raises a question - what does it mean? To me personally the name is lacking meaning, without reading a doc I would struggle to guess what the purpose of the API. What would happen if "try" fails?

Also it is an established pattern to provide API "pairs"  (for the lack of a better term), i.e. if there is an `Open()` method, there should be a `Close()` method. 
Following this principle, if there is `TryFreezeAsync` we now need to have `TryUnfreezeAsync()`, which makes it four in total:
```cs
Task FreezeAsync();
Task<bool> TryFreezeAsync();
Task UnfreezeAsync();
Task<bool> TryUnfreezeAsync();
```

...but this doesn't make sense.

I guess the desired would be having two methods, which form a "pair" and their names being sufficiently informative about the purpose, so developers could reason about those without needing to go and read docs.
Maybe something like 
```cs
Task<bool> SuspendAsync(); // SuspendTabAsync, etc.
Task<bool> ResumeAsync();
```

Reply:
Thanks for the feedback. We'll keep the proposed spec as it is and complete this initial spec PR and forward the questions to the formal API review PR review. Will provide link to that PR here when it is created.

I think that the questions are:
 1.	If we have a TryMethod method, do we also have to have the non-try version Method.
We did some search, and it seems that both WinRT and .Net has some APIs that only has TryXXX without XXX, some .Net samples:
XmlBinaryWriterSession.TryAdd
SequenceReader<T>.TryAdvanceTo
System.Dynamic.DynamicObject has a bunch of TryXXX without XXX
System.Uri.TryCreate
HttpContent.TryComputeLength
TaskScheduler.TryDequeue
TaskScheduler.TryExecuteTask(Inline)
Int32.TryFormat (and all the other types)
UniqueId.TryGetGuid

2.	If we have a paired method Open/Close or Freeze/Unfreeze as in our case, should the pair of methods match for Try and for Async?
We've found both patterns, like for FileRandomAccessStream.OpenAsync, + Close and AppWindow.TryShowAsync + CloseAsync.

3, What name should we use for the API, freeze/unfreeze or suspend/resume?
We feel that suspend/resume are overloaded terms and could potentially confuse developers. We'll leave the name as it and let the formal API review to decide the name of the API.

Reply:
The formal review PR is at: https://github.com/MicrosoftEdge/WebView2Feedback/pull/698